packages/serverpod_shared/lib/src/exceptions/password_missing_exceptions.dart

```diff
@@ -0,0 +1,34 @@
+/// Base class for all password missing exceptions.
+abstract class PasswordMissingException implements Exception {
+  /// The name of the password that is missing.
+  final String passwordName;
+
+  /// The name of the environment variable that is missing.
+  final String envVarName;
+
+  /// Creates a new instance of [PasswordMissingException].
+  PasswordMissingException(this.passwordName, this.envVarName);
+
+  /// The user-friendly message of the exception.
+  String get message => 'Missing password for "$passwordName". '
+      'Please check your config/passwords.yaml file or the '
+      '$envVarName environment variable.';
+
+  @override
+  String toString() => '$runtimeType: The password "$passwordName" or '
+      'the environment variable "$envVarName" is missing.';
+}
+
+/// Exception thrown when a database password is missing from the configuration
+/// or the environment variable.
+final class DatabasePasswordMissingException extends PasswordMissingException {
```

Comment:
Are these subclasses worth it?

I think for the auth module I would just stick to the base one (okay I can't since it's abstract), as I don't see what other useful information I could add.

The only thing that comes to mind would be specifying the `module` that threw this, but then again the affected one should be very clear from the password entry's name. (Even more so in the new modules, with their `$module_$key` pattern.)